[Bug]: 7.0.0-rc.8 Cannot find module 'lib/types/dist' in @storybook/html

### Describe the bug

In Storybook 7.0.0-rc.8 I see a TypeScript compile error when I try to build a project that imports types from `@storybook/html`:
```
../../node_modules/@storybook/html/dist/types-2ec88714.d.ts:3:33 - error TS2307: Cannot find module 'lib/types/dist' or its corresponding type declarations.

import * as lib_types_dist from 'lib/types/dist';
```
(There is a similar issue with `import * as lib_docs_tools_dist from 'lib/docs-tools/dist';`)

If I look in `node_modules` I don't see any directory for `lib` or any entry points named `lib` in `package.json`. Are these standard ES module imports? Is a certain bundler configuration required for this import to be found? If I locally modify `node_modules/@storybook/html/dist/types-2ec88714.d.ts` as described below, the build succeeds.

`import * as lib_types_dist from 'lib/docs-tools/dist';` change to 
`import * as lib_types_dist from '@storybook/docs-tools';`

### To Reproduce

Currently [failing the build](https://github.com/ni/nimble/actions/runs/4536747276/jobs/7993806135#step:9:92) in [this branch of the ni/nimble repository](https://github.com/ni/nimble/tree/users/jattas/storybook-7-plus-typescript). To build, clone that branch then run `npm i` and `npm run build`. The build for `/packages/nimble-components` will fail with this error.

If a smaller reproduction case would be helpful I can try to create one. 

### System

```shell
Environment Info:

  System:
    OS: macOS 13.2.1
    CPU: (10) arm64 Apple M1 Max
  Binaries:
    Node: 16.15.1 - /usr/local/bin/node
    npm: 8.11.0 - /usr/local/bin/npm
  Browsers:
    Chrome: 111.0.5563.110
    Firefox: 111.0.1
    Safari: 16.3
```


### Additional context

A workaround is to change `tsconfig.json` to include `"skipLibCheck": true`, which tells the TypeScript compiler not to run type checking on code in `node_modules`. But that impacts compile for all dependencies so we'd prefer not to have to do this.

I've starting seeing an additional related error in 7.0.4:
```
../../node_modules/@storybook/addon-actions/dist/decorator.d.ts:1:71 - error TS2307: Cannot find module 'lib/preview-api/dist/storybook-channel-mock-a19cd233' or its corresponding type declarations.

1 import * as lib_preview_api_dist_storybook_channel_mock_a19cd233 from 'lib/preview-api/dist/storybook-channel-mock-a19cd233';
```

The workaround we've adopted is to use [patch-package](https://github.com/ds300/patch-package) to run during `npm postinstall` and replace `lib` with `@storybook` in the broken files in `node_modules`. It's a bit annoying to maintain since the files in `node_modules` seem to change names between releases, but at least it allows us to keep `skipLibCheck: false` so we don't have to turn off type checking for all of our other dependencies.

--- a/code/addons/actions/src/decorator.ts
+++ b/code/addons/actions/src/decorator.ts
@@ -1,5 +1,6 @@
 import { global } from '@storybook/global';
 import { useEffect, makeDecorator } from '@storybook/preview-api';
+import type { Addon_DecoratorFunction } from '@storybook/types';
 import { actions } from './runtime/actions';
 
 import { PARAM_KEY } from './constants';
@@ -51,7 +52,7 @@ const applyEventHandlers = (actionsFn: any, ...handles: any[]) => {
   }, [root, actionsFn, handles]);
 };
 
-export const withActions = makeDecorator({
+export const withActions: Addon_DecoratorFunction = makeDecorator({
   name: 'withActions',
   parameterName: PARAM_KEY,
   skipIfNoParametersOrOptions: true,

